src/components/Reports/ReportsBox/ReportsBox.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { InjectAppServices } from '../../../services/pure-di';
+import { FormattedMessage } from 'react-intl';
+import './ReportsBox.css';
+
+class ReportsBox extends React.Component {
+  constructor({ dependencies: { datahubClient } }) {
+    super();
+
+    /** @type { import('../../services/datahub-client').DatahubClient } */
+    this.datahubClient = datahubClient;
+
+    this.state = {
+      visits: 0,
+      dateFrom: null,
+      dateTo: new Date().toLocaleDateString(),
+    };
+
+    this.getVisitsByPeriod = this.getVisitsByPeriod.bind(this);
+  }
+
+  async getVisitsByPeriod(domainName, periodSelectedDays) {
+    let dateFrom = new Date();
+    dateFrom.setDate(dateFrom.getDate() - parseInt(periodSelectedDays));
+    const visits = await this.datahubClient.getVisitsByPeriod(domainName, dateFrom);
+    if (visits) {
+      this.setState({
+        visits: visits,
+        dateFrom: new Date(dateFrom).toLocaleDateString(),
+      });
+    }
```

Comment:
I think that you should always update it, even when there are no visits because in another way the results will not be related to current filters.

Reply:
Oh, `!visits` is not an empty list, is a null array, in that case, it should not happen, and if it happens it is an error. 

What do you think about showing the error in the box?

Reply:
changed